docs/testplan/PMON-Services-Daemons-test-plan.md

```diff
@@ -0,0 +1,395 @@
+         * chassisd : "/usr/share/sonic/platform/chassisdb.conf"
+         * lm-sensors : "/usr/share/sonic/platform/sensors.conf"
+         * fancontrol : "/usr/share/sonic/platform/fancontrol"
+      - Add the daemon to "skipped_daemons" list if the platform configuration file doesn't exist for a daemon to start.
+
+   d. Verify the expected daemon list from the output of a, b, and c with "/etc/supervisor/conf.d/supervisord.conf"
+      - Check the output of `sudo docker exec pmon bash -c '[ -f /etc/supervisor/conf.d/supervisord.conf ] \
+      && cat /etc/supervisor/conf.d/supervisord.conf | grep program'`
+      - Compare the output with the results from the above steps, which is {"all_daemons" - "skipped_daemons"}
+      - Report an error if they don't match
+      - Save the {"all_daemons" - "skipped_daemons"} to "expected_daemons"
+      ```
+   admin@sonic:~$ sudo docker exec pmon bash -c '[ -f /etc/supervisor/conf.d/supervisord.conf ]  && cat /etc/supervisor/conf.d/supervisord.conf | grep program'
+   [program:rsyslogd]
+   [program:lm-sensors]
+   [program:pcied]
+   [program:psud]
+   [program:syseepromd]
+   [program:xcvrd]
+      ```
+   
+   e. Verify each daemon status using `sudo docker exec pmon supervisorctl status`
+      - compare the output with 
+      - Example) "skip_ledd"==true && "skip_thermalctld"==true && none of three config files in c. available
+      ```
+   admin@sonic:~$ sudo docker exec pmon supervisorctl status
+   supervisor-proc-exit-listener    RUNNING   pid 24, uptime 1:30:24
+   dependent-startup                EXITED    Jan 25 08:23 AM
+   rsyslogd                         RUNNING   pid 27, uptime 1:30:22
+   pcied                            RUNNING   pid 36, uptime 1:30:20
+   psud                             RUNNING   pid 34, uptime 1:30:20
+   syseepromd                       RUNNING   pid 35, uptime 1:30:20
+   xcvrd                            RUNNING   pid 32, uptime 1:30:20
+      ```
+
+3. Perform the following steps for the list of the expected daemon
+	a. Verify the daemon is running and check if all expected DB data is populated
+   b. Disable the critical process autorestart if it's enabled using [disable_and_enable_autorestart](https://github.com/Azure/sonic-mgmt/blob/73830827460750d88ba0a5de71bb5c79ad33c46e/tests/process_monitoring/test_critical_process_monitoring.py#L37)
+	c. Stop the daemon using `sudo docker exec pmon supervisorctl stop {daemon}`
+   d. Verify the daemon is not running using `sudo docker exec pmon supervisorctl status {daemon}` and check if all DB data is cleared 
+	e. Restart the daemon using `sudo docker exec pmon supervisorctl start {daemon}`
+   f. Verify the daemon is running using `sudo docker exec pmon supervisorctl status {daemon}` and check if all expected DB data is populated
+   h. Repeat from step c to step f for all exit signals which the daemon can exit with
+   i. Enable the critical process autorestart
```

Comment:
These are common test steps that we want to perform on all State DB-populating daemons (chassisd, pcied, psud, syseepromd, thermalctld, xcvrd). I think you could rework this section so that it describes the common test procedure for every State DB-populating daemon, and in the section below, for those daemons, refer to this procedure and simply specify the data that will be verified for that daemon along with any daemon-specific testing we want to perform outside of this procedure.